core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -254,6 +292,22 @@ private static boolean isSystemPartitioning(PartitioningHandle partitioning)
         return partitioning.getConnectorHandle() instanceof SystemPartitioningHandle;
     }
 
+    private static boolean isScalingPartitioning(PartitioningHandle partitioning)
+    {
+        return partitioning.getConnectorHandle() instanceof ScaleWriterPartitioningHandle;
```

Comment:
`SCALED_WRITER_HASH_DISTRIBUTION` should also be of `ScaleWriterPartitioningHandle` type